admin/sql/updates/20200630-mbs-10921-unlist-unrelated-edits.sql

```diff
@@ -0,0 +1,286 @@
+\set ON_ERROR_STOP 1
+BEGIN;
+
+SET LOCAL statement_timeout TO '15min';
+
+-- I) Artists
+
+\echo MBS-10921 (1/12): Unlist “Add relationship” edits from artists’ history...
+DELETE FROM edit_artist
+WHERE (edit, artist) IN (
+    SELECT ea.edit, ea.artist
+    FROM edit_artist ea
+    JOIN edit e ON ea.edit = e.id
+    JOIN edit_data ed ON e.id = ed.edit
+    WHERE e.id >= 70658947 AND e.id < 71006471
+    AND e.type = 90
+    AND jsonb_extract_path_text(ed.data, 'type0') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'type1') = 'work'
+    AND jsonb_extract_path_text(ed.data, 'entity0', 'id')::int IN (
+        SELECT id FROM recording
+    )
+    AND jsonb_extract_path_text(ed.data, 'entity0', 'id')::int NOT IN (
+        SELECT r.id
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
+    LIMIT 2000 -- of 208074
+)
+RETURNING *;
+
+\echo MBS-10921 (2/12): Unlist “Edit relationship” edits from artists’ history...
+DELETE FROM edit_artist
+WHERE (edit, artist) IN (
+    SELECT ea.edit, ea.artist
+    FROM edit_artist ea
+    JOIN edit e ON ea.edit = e.id
+    JOIN edit_data ed ON e.id = ed.edit
+    WHERE e.id >= 70658947 AND e.id < 71006471
+    AND e.type = 91
+    AND jsonb_extract_path_text(ed.data, 'type0') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'type1') = 'work'
+    AND jsonb_extract_path_text(ed.data, 'link', 'entity0', 'id')::int IN (
+        SELECT id FROM recording
+    )
+    AND jsonb_extract_path_text(ed.data, 'link', 'entity0', 'id')::int NOT IN (
+        SELECT r.id
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
```

Comment:
https://github.com/metabrainz/musicbrainz-server/blob/4d3b7779ffd1d6753cd784fce21cd6f5b0cd3e8b/lib/MusicBrainz/Server/Edit/Relationship/Edit.pm#L267-L269 also output `$old->{entity0}` and `$new->{entity0}` (which would be present if the recording changed). Though `$old->{entity0}` should always be equal to `$link->{entity0}` if present. Maybe something like this?
```suggestion
    AND EXISTS (
        SELECT 1
          FROM recording
         WHERE id IN (VALUES
             (jsonb_extract_path_text(ed.data, 'link', 'entity0', 'id')::int),
             (jsonb_extract_path_text(ed.data, 'new', 'entity0', 'id')::int))
    )
    AND ea.artist NOT IN (
        SELECT acn.artist
          FROM artist_credit_name acn
          JOIN recording r ON r.artist_credit = acn.artist_credit
         WHERE r.id IN (VALUES
             (jsonb_extract_path_text(ed.data, 'link', 'entity0', 'id')::int),
             (jsonb_extract_path_text(ed.data, 'new', 'entity0', 'id')::int))
    )
```

Reply:
Picked up as https://github.com/metabrainz/musicbrainz-server/pull/1581/commits/9eb0f6b311933a8e7c003020c1ad4a14a537d6f8, amended with https://github.com/metabrainz/musicbrainz-server/pull/1581/commits/46ff2a3265a23d7e9b4339297a5151812576d39f.

Reply:
That is actually unneeded, reverted in https://github.com/metabrainz/musicbrainz-server/pull/1581/commits/29074454c43a5123ee86f3a71d75594f5ac0fdd5.